native/cocos/bindings/jswrapper/PrivateObject.h

```diff
@@ -185,25 +187,18 @@ inline void inHeap(void *ptr) {
 }
 #endif
 
-template <typename T>
-typename std::enable_if<std::is_base_of<cc::RefCounted, T>::value, PrivateObjectBase *>::type
-cc_tmp_new_ptr(T *cobj) {
-    return ccnew CCSharedPtrPrivateObject<T>(cc::IntrusivePtr<T>(cobj));
-}
-template <typename T>
-typename std::enable_if<!std::is_base_of<cc::RefCounted, T>::value, PrivateObjectBase *>::type
-cc_tmp_new_ptr(T *cobj) {
-    return ccnew SharedPrivateObject<T>(std::shared_ptr<T>(cobj));
-}
-
 template <typename T>
 inline PrivateObjectBase *make_shared_private_object(T *cobj) { // NOLINT
     static_assert(!std::is_same<T, void>::value, "void * is not allowed");
 // static_assert(!std::is_pointer_v<T> && !std::is_null_pointer_v<decltype(cobj)>, "bad pointer");
 #if CC_DEBUG
     inHeap(cobj);
 #endif
-    return cc_tmp_new_ptr(cobj);
+    if constexpr (std::is_base_of<cc::RefCounted, T>::value) {
+        return ccnew CCSharedPtrPrivateObject<T>(cc::IntrusivePtr<T>(cobj));
```

Comment:
Should we rename `CCSharedPtrPrivateObject ` to `CCInstrusivePtrPrivateObject `?

Reply:
yes, it is more precise